elytron/WFCORE-5744-ssl-certificate-expiration-warning.adoc

```diff
@@ -0,0 +1,218 @@
+promoted-by:
+---
+= Display warning for expiring ssl ceritificates
+:author:            Bartosz Baranowski
+:email:             bbaranow@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+__<The entire document should be one to two pages long. We will write each analysis document as if it is a conversation with a future developer. This requires a good writing style, with full sentences organized into paragraphs. Bullets are acceptable only for visual style, not as an excuse for writing sentence fragments.>__
+
+== Overview
+
+Goal of this RFE is two fold, to formalize and improve ways to get feedback on state of SSL certificates in server and to add periodic check, based on configuration.
+
+Periodic check is more or less self explanatory. Currently upon startup elytron subsystem will will print out information about certificates that expired. There are few problems with this. Firstly its exception driven and lacks a bit of information( ie keystore). Secondly its one-off on startup, this operation potentially should be performed periodically.
+
+Secondly currently there exist dedicated command _should-renew-certificate(date_date_to_check)_ . It would be good to provide such detail along with _read-alias/s_ command output.
+The _read-alias_ will will provide health indication with fixed TTL( of 7 days at least), while _should-renew-certificate_ will allow detailed inerogration on certificate health.
+
+
+=== User Stories
+
+Long runing server, with periodic information on certificates will allow admins to not only setup tools to warn them about potential security risk, but also schedule maintanence window based on this.
+
+== Issue Metadata
+
+
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1863[EAP7-1863]
+* https://issues.redhat.com/browse/WFCORE-5744[WFCORE-5744]
+
+=== Affected Projects or Components
+
+ * Elytron integration
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+ * create output format that is not driven by exception, so possibly it can be easily digested via log munching, mail notification or any other mechanism users see fit to employ over log handler.
+ * read-alias command will have additional KVP "Validity=[valid, EXPIRED, about to expire, not yet]"
+ ** Validity semantics:
+ *** not yet - ceritifacete is still too young
+ *** EXPIRED - current date is after expiration
+ *** about to expire - certificate is still valid. If certificate lifespan is week or smaller, this rougly corresponds to day to EOL, otherwise 7 days to EOL
+ *** valid - more than above boundries
+ * add config options to handle:
+ ** frequency of warning
+ *** 0 - turn off - one time check on startup as in previous version
+ *** n+ - millisecconds between periodic checks
+
+
```

Comment:
If we are going to do some periodic checks I think validity intervals may need to be configurable, I can imaging different users having different notification requirements.